analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.IsNull.cs

```diff
@@ -89,8 +87,31 @@ public void IsNull_Coalesce_UnknownToUnknown()
 Tag(""UnknownToUnknown"", unknownToUnknown);";
             var validator = SETestContext.CreateCS(code, ", object arg1, object arg2").Validator;
             validator.ValidateContainsOperation(OperationKind.IsNull);
-            validator.TagValues("UnknownToUnknown").Should().HaveCount(1)
-                .And.ContainSingle(x => x == null);
+            validator.TagValues("UnknownToUnknown").Should().SatisfyRespectively(
+                x => x.Should().HaveNoConstraints(),
+                x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+        }
+
+        [TestMethod]
+        public void IsNull_Coalesce_AssignedToVariable()
+        {
+            const string code = """
+                var value = arg ?? "N/A";
+                Tag("Value", value);
+                """;
+            var validator = SETestContext.CreateCS(code, ", string arg").Validator;
+            validator.ValidateTag("Value", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void IsNull_Coalesce_AssignedToSelf()
+        {
+            const string code = """
+                arg = arg ?? "N/A";
+                Tag("Arg", arg);
+                """;
+            var validator = SETestContext.CreateCS(code, ", string arg").Validator;
+            validator.ValidateTag("Arg", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
         }
```

Comment:
Consider adding some tests with nullable bool so we see what states are produced from that:
`var value = nullableBoolParam1 ?? nullableBoolParam2 ?? true;`. It should be something like this, right?

| nullableBoolParam1 | nullableBoolParam2 | value |
| --- | ---| --- |
| **true** | true | true |
| **true** | false | true |
| **true** | null | true |
| **false** | null | false |
| **false** | true | false |
| **false** | false | false |
| null | **true** | true |
| null | **false** | false |
| null | null | **true** |

Reply:
I'll do another PR for this.